src/types.ts

```diff
@@ -16,14 +16,24 @@ export interface Params {
   readonly [key: string]: object | Primitive | undefined | null
 }
 
+export interface NestedParam {
+  // We need the NestedParamArray here for circulatiry
+  // eslint-disable-next-line no-use-before-define
+  [param: string]: Primitive | undefined | null | NestedParam | NestedParamArray
+}
+
+// Eslint will try to fix this to a type, but we need it as an interface for the recursability
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+export interface NestedParamArray extends Array<Primitive | NestedParam | NestedParamArray> {}
+
 export interface RequestParams {
   readonly auth?: Auth
   readonly body?: object | string
   readonly headers?: Headers
   readonly host?: string
   readonly params?: Params
   readonly timeout?: number
-  [param: string]: object | Primitive | undefined | null
+  [param: string]: object | Primitive | undefined | null | NestedParam | NestedParamArray
```

Comment:
Even though, in theory, all `object` type necessities should be covered by `NestedParam` here, we need it explicit here, otherwise `body`'s type becomes incompatible